uat/testing-features/src/main/java/com/aws/greengrass/steps/CloudWatchSteps.java

```diff
@@ -41,13 +48,31 @@ public CloudWatchSteps(
             TestContext testContext,
             AWSResourcesContext resourcesContext,
             AWSResources resources,
-            WaitSteps waitSteps
+            WaitSteps waitSteps,
+            CloudWatchLogsClient cwClient
     ) {
         this.logsLifecycle = logsLifecycle;
         this.testContext = testContext;
         this.resourceContext = resourcesContext;
         this.resources = resources;
         this.waitSteps = waitSteps;
+        this.cwClient = cwClient;
+    }
+
+    @Then("I delete the log group of type {word} for component {word} if it exists")
+    public void deleteLogGroup(String componentType, String componentName) {
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy/MM/dd", Locale.ENGLISH);
+        formatter.setTimeZone(TimeZone.getTimeZone("UTC")); // All dates are UTC, not local time
+
+        String region = resourceContext.region().toString();
+        String logGroupName = String.format("/aws/greengrass/%s/%s/%s", componentType, region, componentName);
+        DeleteLogGroupRequest request = DeleteLogGroupRequest.builder().logGroupName(logGroupName).build();
+
+        try {
+            this.cwClient.deleteLogGroup(request);
+        } catch (ResourceNotFoundException notFound) {
+            logger.atDebug().kv("groupName", logGroupName).cause(notFound).log("Failed to delete group");
```

Comment:
Is this the only type of exception for deleteLogGroup? Nit: `Failed to delete log group`